app/services/parse/google_parse_service.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Parse
+  class GoogleParseService
+    SELECTORS = {
+      adwords_top: 'div.pla-unit-container a.pla-unit-title-link',
+      adwords_body: 'div.data-text-ad div.v5yQqb',
+      non_adwords: 'div#search div.yuRUbf a[data-ved]',
+      href: 'href'
+    }.freeze
+
+    def initialize(html_code)
+      @html_code = html_code
+      raise IcRailsHueyToby::Errors::ParseServiceError unless html_code
+    end
+
+    def parse
+      parse_html
+    end
+
+    private
+
+    attr_reader :html_code
+
+    # rubocop:disable Metrics/MethodLength
+    def parse_html
+      doc = Nokogiri::HTML(html_code)
+
+      adwords_top_urls = parse_top_adwords(doc)
+      adwords_body_urls = parse_body_adwords(doc)
+      non_adwords_urls = parse_non_adwords(doc)
+
+      adwords_top_count = adwords_top_urls.count
+      adwords_body_count = adwords_body_urls.count
+      adwords_total_count = adwords_top_count + adwords_body_count
+      non_adwords_count = non_adwords_urls.count
+      total_links_count = adwords_total_count + non_adwords_count
```

Comment:
Should we split these blocks into 2 methods so that we don't need to add `rubocop:disable Metrics/MethodLength`?

Reply:
We can definitely and should break into parsing adword related and non-adwords content. The total links parsing can also be another method.

Reply:
Fixed [c8e81d8](https://github.com/nimblehq/ic-rails-huey-toby/pull/37/commits/c8e81d88c67f79a754298d61aa528cd44d75cec9)